### PR TITLE
LibriMix

### PR DESCRIPTION
**Pull request by JorisCos**
### What is this PR about ? 
We introduce a new open source dataset derived from LibriSpeech and WHAM for source separation.
In this PR, we give all the content needed to generate mixtures and sources with any number of speakers. For now, clean and noisy mixtures are available.

### Content

-  The scripts folder contains the scripts to generate metadata from LibriSpeech, WHAM and LibriMix and the actual dataset with sources and mixtures. The dataset organization is meant to match WHAM's organization.

-  The metadata folder contains metadata for Librispeech, WHAM and LibriMix. Only the metadata from LibriMix are needed to generate the dataset

- Finally, we give a bash script to make the all installation easier.
 
@mpariente can you make a review to be sure that the generation code fit asteroid's standard

### CODE REVIEW

**Comment by mpariente** [comment]
### What I'd do :
- Remove `parse_options.sh` and `prepare_python_env.sh`
- Just replace `run.sh` by a simple bash script (`generate_librimix.sh`) which takes only two argument (with `$1` and `$2`, not with the `--options` mechanism), the storage directory and the number of sources. 
- In the storage directory, download Librispeech into $storage_dir/LibriSpeech
- Create Libri{n_src}Mix in $storage_dir/Libri{n_src}Mix
- Add a `requirements.txt` in the repo and ask the user to install it in the README.
- I would rename Dataset_generation to `scripts`, at least all lowercase. 

Maybe, by default, generate all the 8k,16k and min,max versions and tell the user it can be changed in the bash script?

File: generate_librimix.sh [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+python scripts/augment_train_noise.py --wham_dir $wham_dir
+python scripts/create_librimix_from_metadata.py --librispeech_dir $librispeech_dir \
+--wham_dir $wham_dir \
+--metadata_dir $metadata_dir \
+--librimix_outdir $librimix_outdir \
+--n_src $n_src \
+--freqs 8k 16k \
+--modes min max \
+--types mix_clean mix_both
```
Comment (mpariente) [inline generate_librimix.sh:51]: To be indented

File: scripts/augment_train_noise.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    augment_noise(sound_paths, 0.8)
+    augment_noise(sound_paths, 1.2)
+
+
+def augment_noise(sound_paths, speed):
+    fx = (AudioEffectsChain().speed(speed))
+    print(f"Change speed with factor {speed}")
+    for sound_path in tqdm(sound_paths, total=len(sound_paths)):
+        s, rate = sf.read(sound_path)
+        s = s[:, 0]
```
[the remark this answers is not in the document]
Comment (JorisCos) [reply scripts/augment_train_noise.py:29]: Yes. We could change speed for all channels, but we always use the first channel as noise and get rid of the other one. Maybe getting rid of it in augmentation process is a bit confusing but it has the advantage of saving space.

File: generate_librimix.sh (hunk 1 above)
Comment (JorisCos) [reply generate_librimix.sh:51]: Thanks

**Pull request merged**